merkle/coniks/coniks.go

```diff
@@ -124,11 +125,19 @@ func (m *hasher) maskIndex(b *bytes.Buffer, index []byte, depth int) {
 		panic(fmt.Sprintf("depth: %d, want <= %d && > 0", got, want))
 	}
 
+	prevLen := b.Len()
 	if depth > 0 {
 		// Copy the first depthBytes.
 		depthBytes := (depth + 7) >> 3
-		b.Write(index[:depthBytes])
+		if depth > 1 {
```

Comment:
Should we add a test for this case?

Reply:
I think it's covered by existing tests but I'll go back and make sure.

Reply:
This was incorrect, should have been `depthBytes`. However for depthBytes to be zero depth must be zero so it worked anyway by luck.

Reply:
Additional tests added as a crosscheck against bit twiddling on a big.Int.